Merchant/App/PaymentInput/PaymentInputViewController.swift

```diff
@@ -39,17 +44,37 @@ class PaymentInputViewController: PinViewController {
     
     var presenter: PaymentInputPresenter?
     
-    var amount: Int = 0
     var amountStr: String = "0"
+   
+    private let notificationCenter: NotificationCenter = NotificationCenter.default
+    
+    @objc private func currencyDidChange(_ notification: Notification) {
+        DispatchQueue.main.async {
+            self.onCleanAmount()
+            self.presenter?.didSetAmount("0")
+            self.currencyLabel.text = UserManager.shared.selectedCurrency.ticker
+        }
+    }
     
     override func viewDidLoad() {
         self.hasValid = true
-        
+        notificationCenter.addObserver(self, selector: #selector(currencyDidChange),
```

Comment:
You do not subscribe in the view controller. You add logic to the view. The view is not responsible for this.

Reply:
If the variables are only and should only be accessible in this file, pls use "fileprivate" instead of "private"